src/vs/editor/browser/controller/textAreaInput.ts

```diff
@@ -210,16 +213,27 @@ export class TextAreaInput extends Disposable {
 		private readonly _host: ITextAreaInputHost,
 		private readonly _textArea: ICompleteTextAreaWrapper,
 		private readonly _OS: OperatingSystem,
-		private readonly _browser: IBrowser
+		private readonly _browser: IBrowser,
+		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
+		@ILogService private readonly _logService: ILogService
 	) {
 		super();
 		this._asyncTriggerCut = this._register(new RunOnceScheduler(() => this._onCut.fire(), 0));
-		this._asyncFocusGainWriteScreenReaderContent = this._register(new RunOnceScheduler(() => this.writeScreenReaderContent('asyncFocusGain'), 0));
-
+		if (this._accessibilityService.isScreenReaderOptimized()) {
+			this._asyncFocusGainWriteScreenReaderContent.value = this._register(new RunOnceScheduler(() => this.writeScreenReaderContent('asyncFocusGain'), 0));
+		}
 		this._textAreaState = TextAreaState.EMPTY;
 		this._selectionChangeListener = null;
-		this.writeScreenReaderContent('ctor');
-
+		if (this._accessibilityService.isScreenReaderOptimized()) {
+			this.writeScreenReaderContent('ctor');
+		}
+		this._register(this._accessibilityService.onDidChangeScreenReaderOptimized(() => {
+			if (this._accessibilityService.isScreenReaderOptimized() && !this._asyncFocusGainWriteScreenReaderContent.value) {
+				this._asyncFocusGainWriteScreenReaderContent.value = this._register(new RunOnceScheduler(() => this.writeScreenReaderContent('asyncFocusGain'), 0));
+			} else {
+				this._asyncFocusGainWriteScreenReaderContent.clear();
+			}
+		}));
```

Comment:
`Event.runAndSubscribe` will help you keep the initialization and event consistent